lib_eio/net.mli

```diff
@@ -187,8 +191,12 @@ val accept_sub :
 
 (** {2 Datagram Sockets} *)
 
-val datagram_socket : sw:Switch.t -> #t -> Sockaddr.datagram -> <datagram_socket; Flow.close>
-(** [datagram_socket ~sw t addr] creates a new datagram socket that data can be sent to
+val datagram_socket :
+     sw:Switch.t
+  -> #t
+  -> [Sockaddr.datagram | `UdpV4 | `UdpV6]
```

Comment:
```suggestion
  -> [< Sockaddr.datagram | `UdpV4 | `UdpV6]
```

Reply:
I was asking Bikal about this on slack, why is the `<` preferred in this case ?

Reply:
If you have a `Sockaddr.datagram` then you should be able to pass it to this function without first having to cast it to allow the extra tags.

Sometimes the function generating the value will have a `[> ...]`, which allows it to adapt there instead, but it's good to be flexible.

Reply:
Done.